1-JavaScript/Week2/homeworkSolition/ex5-sol-shoppingCartPure.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+function addToShoppingCart( shoppingCartNew, groceryItem) {
+ const shoppingCart= [...shoppingCartNew]
+  if(groceryItem !== undefined){
+    shoppingCart.push(groceryItem)
+  }
+  if(shoppingCart.length > 3 ){
+  shoppingCart.shift()
+   
+  }
+  console.log(`   the items bought are ${shoppingCart.join(", ")}!`)
+  return shoppingCart
+}
```

Comment:
Almost there 😄 

The console.log is not necessary and the spread syntax needs to be used for making the new array 🙂 

```suggestion
function addToShoppingCart(shoppingCart, groceryItem) {
 let newShoppingCart= []
 
  if(groceryItem !== undefined){
     newShoppingCart = [...shoppingCart, groceryItem]
  }
  
  if(newShoppingCart.length > 3 ){
     newShoppingCart.shift()
  }

  return newShoppingCart
}
```